### PR TITLE
Removed InputHandler field from Window class

### PR DESCRIPTION
**Pull request by Soskar1**
I made small changes to the Window class code. Removed InputHandler field from Window class.
Now we need to create InputHandler object separately and add listeners by ourselves. It gives more control over input registration.

In the future plans, I will refactor Input class. I think we can move the input queue elsewhere. I will try to experiment with this in another branch

### CODE REVIEW

File: examples/core/input/0_input_keyboard/main.cpp [hunk 1]
```diff
@@ -18,14 +18,18 @@ int main() {
         std::cout << "key press\n";
     };
 
-    auto input = window->get_input();
+    InputHandler input_handler = InputHandler();
```
Comment (Maksasj) [inline examples/core/input/0_input_keyboard/main.cpp:21]: Same as https://github.com/Maksasj/bebone/pull/123#discussion_r1660209425

File: examples/gfx/opengl/5_opengl_flappy_bird/main.cpp [hunk 2]
```diff
@@ -27,8 +27,12 @@ int main() {
     glEnable(GL_BLEND);
     glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
 
-    auto input = window->get_input();
-    Game game(screen_width, screen_height, input);
+    InputHandler input_handler = InputHandler();
```
Comment (Maksasj) [inline examples/gfx/opengl/5_opengl_flappy_bird/main.cpp:30]: Same as above

**Review by Maksasj — CHANGES_REQUESTED** [review]

File: examples/core/input/0_input_keyboard/main.cpp (hunk 1 above)
Comment (Soskar1) [reply examples/core/input/0_input_keyboard/main.cpp:21]: That's my code style. I don't think it harms our code in any way. For me, the `auto` keyword reduces readability. I don't like to use it, except when there is `std::shared_pointer`.

File: examples/gfx/opengl/5_opengl_flappy_bird/main.cpp (hunk 2 above)
Comment (Soskar1) [reply examples/gfx/opengl/5_opengl_flappy_bird/main.cpp:30]: That's my code style. I don't think it harms our code in any way. For me, the `auto` keyword reduces readability. I don't like to use it, except when there is `std::shared_pointer`.

**Review by Maksasj — APPROVED** [review]

**Pull request merged**